### PR TITLE
[SSH] SSH support including ATs bound to keys

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: msal/token_cache.py [hunk 1]
```diff
@@ -165,6 +166,10 @@ def __add(self, event, now=None):
                     "expires_on": str(now + expires_in),  # Same here
                     "extended_expires_on": str(now + ext_expires_in)  # Same here
                     }
+
+                if (at.get("token_type").casefold() == "ssh-cert"):
```
[the remark this answers is not in the document]
Comment (bgavrilMS) [reply msal/token_cache.py]: I did a bit of testing around this and it seems that - argument names are case insensitive but argument values are case sensitive. So 

`"toKEN_type": "ssh-cert" ` - works
`"token_type": "SSH-cert"` - fails

[the remark this answers is not in the document]
Comment (bgavrilMS) [reply msal/token_cache.py]: So I think the changes you've made to remove `casefold` are good, we can expect case sensitivity everywhere. 

Ship it!

[the remark this answers is not in the document]
Comment (rayluo) [reply msal/token_cache.py]: For the sake of completeness, it turns out there is a much deeper rabbit hole here for the case-sensitivity of the "token_type" value.

1. In general, the common sense should always be, "**Unless otherwise noted, all the protocol parameter names and values  are case sensitive.**" - Quoted from [section 1.9.  Notational Conventions of the OAuth2 RFC](https://tools.ietf.org/html/rfc6749#section-1.9). Even the OIDC specs (which is largely designed on top of OAuth2) considers that "token_type" value to be case sensitive, by saying "The OAuth 2.0 token_type response parameter value MUST be Bearer", in [section 3.1.3.3.  Successful Token Response](https://openid.net/specs/openid-connect-core-1_0.html#TokenResponse). If the story ends here, everyone will live happily ever after.

2. But, later in same [OAuth2 RFC, section 4.2.2.  Access Token Response](https://tools.ietf.org/html/rfc6749#section-4.2.2), it says:

   > **token_type**
         REQUIRED.  The type of the token issued as described in
         Section 7.1.  **Value is case insensitive.**

3. And then I found this 4-year-old issue. Michael Jones the specs writer [changed the OIDC specs](https://bitbucket.org/openid/connect/issues/985/use-bearer-in-token_type-in-implicit-flow) this year to clarify that "token_type value is case insensitive", although that change does not yet seem to be published.

What do all these mean to this PR? Nothing. Coincidently, in our latest commit, we already [refactored that line to not rely on the token_type value at all](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/102/files#diff-6f83eec40714321407e16ebed558f014R169).

(We do have [another line still using lower case "ssh-cert"](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/102/files#diff-f7ae931bcdab4b14f3a256baf7d6922aR523), but that line is for token request, not for token response. So the No. 2 and No. 3 points above do not apply. Besides, @bgavrilMS already tested it that our service indeed expects a lower case "ssh-cert" as input.)

TL; DR: Ship it! :-)
